data/collector_test.go

```diff
@@ -264,23 +264,24 @@ func TestCtxCancelledNotLoggedInTickerBasedCaptureAfterClose(t *testing.T) {
 
 	params := CollectorParams{
 		ComponentName: "testComponent",
-		// Ensure that we use ticker-based capture since capturing at 1000+ Hz
-		// uses a sleep-based capture that may let a context cancelation error
-		// occasionally be logged.
-		Interval:     sleepCaptureCutoff + time.Microsecond,
-		MethodParams: map[string]*anypb.Any{"name": fakeVal},
-		Target:       target,
-		QueueSize:    queueSize,
-		BufferSize:   bufferSize,
-		Logger:       logger,
+		Interval:      time.Nanosecond,
```

Comment:
The test doesn't need to be committed this way -- this was just what I ran to get much more reliable failures.